WordPress/gutenberg
# Site editor sidebar: abstracting footer so it can be used across details screens

Maybe resolves https://github.com/WordPress/gutenberg/issues/51343


## What?
- Consolidates the footer component for the site editor sidebar (the styles panel was formerly using it's own)
- Injects a link to the `revision.php` diff where a `predecessor-version` exists in the entity record (pages, templates)

## Why?
The great unification! Seriously, less stuff to maintain. 

# How?

Don't you mean, Wow?!

Where a `predecessor-version` exists, we'll build the link to `revision.php` unless an `onClick` handler is pass, which is the case with global styles.

If none of these exist, and there's a last modified entry, just show that info.

## Testing Instructions
Save some changes to a page, a template, a pattern and global styles. Enough so you have a few revisions under your belt.

Open the site editor and browse the relevant items in the sidebar.

Check that the "Last modified" footer item looks and works fine for each one. 

Click on the links and ensure they either go to `revision.php` or to some other thing, like the Style panel does.


## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/6458278/ca0aadc0-7395-4a6d-89a6-46092a6b7468
